x/oracle/keeper/grpc_query.go

```diff
@@ -0,0 +1,204 @@
+package keeper
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/umee-network/umee/x/oracle/types"
+)
+
+// Querier implements a QueryServer for the x/oracle module
+type querier struct {
+	Keeper
+}
+
+// NewQuerier returns an implementation of the oracle QueryServer interface
+// for the provided Keeper.
+func NewQuerier(keeper Keeper) types.QueryServer {
+	return &querier{Keeper: keeper}
+}
+
+var _ types.QueryServer = querier{}
+
+// Params queries params of x/oracle module
+func (q querier) Params(
+	c context.Context,
+	req *types.QueryParamsRequest,
+) (*types.QueryParamsResponse, error) {
+	ctx := sdk.UnwrapSDKContext(c)
+	params := q.GetParams(ctx)
+
+	return &types.QueryParamsResponse{Params: params}, nil
+}
+
+// ExchangeRates queries exchange rates of all denoms
```

Comment:
nit: Same as above. Change the comment to explain that if specified it will return the exchange rate for a single denom instead of all.